plugin/SearchBarItem.qml

```diff
@@ -162,11 +184,14 @@ BackgroundItem {
         EnterKey.onClicked: {
             if (text != "") {
                 _searchText = text
+                focus = false // Need to be called after saving text
+                              // but before the request, see the code
+                              // in onActiveFocusChanged, when activeFocus is false.
```

Comment:
> it either searches or just dismisses. 

Yes, I can reproduce. I've tried to correct this also. I've follow your advice, and simplified the logic there, using _searchText for the request (don't know why I didn't do it before…), even using a connection on_SearchTextChanged to emit the request indeed.